dot/simswap/util/reverse2original.py

```diff
@@ -103,22 +98,21 @@ def reverse2wholeimage(
     use_mask=False,
     use_gpu=True,
 ):
-
     target_image_list = []
     img_mask_list = []
+    device = torch.device("cuda" if use_gpu else "cpu")
     if use_mask:
-        if use_gpu:
-            smooth_mask = SoftErosion(
-                kernel_size=17, threshold=0.9, iterations=7
-            ).cuda()
-        else:
-            smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).cpu()
+        smooth_mask = SoftErosion(kernel_size=17, threshold=0.9, iterations=7).to(
+            device
+        )
     else:
         pass
 
     for swaped_img, mat, source_img in zip(swaped_imgs, mats, b_align_crop_tenor_list):
         swaped_img = swaped_img.cpu().detach().numpy().transpose((1, 2, 0))
```

Comment:
can these 3 operations be compressed? is the conversions to numpy necessary?

Reply:
Fixed